CRM/Contribute/BAO/Query.php

```diff
@@ -937,11 +778,11 @@ public static function defaultReturnProperties($mode, $includeCustomFields = TRU
         // torture
         'accounting_code' => 1,
         // small
-        'payment_instrument' => 1,
+        // 'payment_instrument' => 1,
         // kittens
         'payment_instrument_id' => 1,
         // argh
-        'check_number' => 1,
+        'contribution_check_number' => 1,
```

Comment:
yep - that's what unique fields does for you. Control your hatred @colemanw  - it does simplify things really. Although I had to add some legacy handling to the api